app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,78 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.models.Task;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/tasks")
+@CrossOrigin
+public class TaskController {
+
+    private final TaskRepository taskRepository = new TaskRepository();
+
+    @GetMapping
+    public List<Task> list() {
+        return taskRepository.findList();
+    }
+
+    @PostMapping
+    public Task create(@RequestBody Task task) {
+        return taskRepository.add(task);
+    }
+
+    @GetMapping("/{id}")
+    public Task detail(@PathVariable Long id, HttpServletResponse response) {
+        Optional<Task> find = taskRepository.findDetail(id);
+
+        if (find.isEmpty()) {
+            response.setStatus(400);
+            return null;
+        }
+
+        return find.get();
+    }
```

Comment:
`400`을 그대로 사용하는 것이 아쉽군요. 이렇게 해보면 어떨까요?
```suggestion
    @GetMapping("/{id}")
    public ResponseEntity<Task> detail(@PathVariable Long id) {
        Optional<Task> find = taskRepository.findDetail(id);

        if (find.isEmpty()) {
            return ResponseEntity.badRequest()
                .build();
        }
        return ResponseEntity.ok(find.get());
    }
```

Reply:
06b5a57 ResponseEntity 같이 재너릭을 사용하는 클래스 중에서
무엇을 지정해야 할 지 모르는 경우는 어떻게 해야 할까요?
예로 들어 TaskController에서 삭제하는 경우 응답 body에 특정 값을 사용하지 않을 때 입니다.

Reply:
만약 `ResponseEntity`를 사용하기로 마음 먹었을 때 타입 지정이 애매하면 `ResponseEntity<?>`를 써도 괜찮아요. 
그리고 이렇게 리턴하면 됩니다.

```java
return ResponseEntity.ok().build();
```

`ResponseEntity`를 사용하는 건 선택이에요. 선택을 해도 되고 안 해도 됩니다. 제가 여기에서 이 방법을 제안드린 건 `return null;`이 영 찝찝하게 보여서 그런 거고요.